nomad/structs/config/vault.go

```diff
@@ -83,6 +84,46 @@ type VaultConfig struct {
 
 	// TLSServerName, if set, is used to set the SNI host when connecting via TLS.
 	TLSServerName string `mapstructure:"tls_server_name"`
+
+	// Servers-only fields.
+
+	// UseIdentity defines if workload identities should be used to derive
+	// Vault tokens.
+	//
+	// It is a transitional field used only during the adoption period of
+	// workload identities and will be ignored and removed in future versions.
+	UseIdentity *bool `mapstructure:"use_identity"`
+
+	// DefaultIdentity is the default workload identity configuration used when
+	// a job has a `vault` block but no `identity` named "vault".
+	DefaultIdentity *WorkloadIdentityConfig `mapstructure:"default_identity"`
+
+	// Deprecated fields.
+
+	// Token is the Vault token given to Nomad such that it can
+	// derive child tokens. Nomad will renew this token at half its lease
+	// lifetime.
+	//
+	// Deprecated: Nomad 1.7.0 is able to derive Vault tokens from workload
+	// identities. This field will be removed in a future release.
+	Token string `mapstructure:"token"`
+
+	// AllowUnauthenticated allows users to submit jobs requiring Vault tokens
+	// without providing a Vault token proving they have access to these
+	// policies.
+	//
+	// Deprecated: Nomad 1.7.0 no longer requires a Vault token for job
+	// operations. This field will be removed in a future release.
+	AllowUnauthenticated *bool `mapstructure:"allow_unauthenticated"`
+
+	// TaskTokenTTL is the TTL of the tokens created by Nomad Servers and used
+	// by the client.  There should be a minimum time value such that the client
+	// does not have to renew with Vault at a very high frequency
+	//
+	// Deprecated: Nomad 1.7.0 derives tokens from workload identities that
+	// receive their TTL configuration from the Vault role used. This field
+	// will be removed in a future release.
+	TaskTokenTTL string `mapstructure:"task_token_ttl"`
```

Comment:
This was not part of the RFC, but looking at the code it seems to only be used in the legacy flow. I may be mistaken though 😅